### PR TITLE
feat: Allow passing a custom signer

### PR DESCRIPTION
**Pull request by salmanm**
Adds a custom signer support. 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,6 +39,12 @@ fastify.get('/', (req, reply) => {
 })
 ```
 
+## Options
+
+- `secret`: A `String` to use as secret to sign the cookie using [`cookie-signature`](http://npm.im/cookie-signature). If you want to implement more sophisticated cookie signing mechanism, you can supply an `Object` instead. Read more about it in [Custom cookie signing](#custom-cookie-signing).
```
Comment (simoneb) [inline README.md:44]: wrong link custom-cookie-signing

**Review by simoneb — APPROVED** [review]
LGTM

File: signer.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const cookieSignature = require('cookie-signature')
```
Comment (mcollina) [inline signer.js:1]: use strict is missing

File: test/cookie.test.js [hunk 3]
```diff
@@ -220,6 +221,36 @@ test('cookies signature', (t) => {
     })
   })
 
+  t.test('custom signer', t => {
```
Comment (mcollina) [inline test/cookie.test.js:224]:
```suggestion
  test('custom signer', t => {
```

File: test/cookie.test.js [hunk 4]
```diff
@@ -244,6 +275,34 @@ test('cookies signature', (t) => {
       t.deepEqual(JSON.parse(res.body), { unsigned: 'foo' })
     })
   })
+
+  t.test('unsignCookie decorator with custom signer', t => {
```
Comment (mcollina) [inline test/cookie.test.js:279]:
```suggestion
  test('unsignCookie decorator with custom signer', t => {
```

File: signer.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+module.exports = function (secret) {
+  const sign = (value) => {
+    return cookieSignature.sign(value, secret)
+  }
+
+  const unsign = (value) => {
+    return cookieSignature.unsign(value, secret)
+  }
+
+  return { sign, unsign }
```
Comment (mcollina) [inline signer.js:12]: I would not use variables but define the function inline

**Review by mcollina — APPROVED** [review]
lgtm

File: README.md (hunk 1 above)
Comment (salmanm) [reply README.md]: Fixed

File: test/cookie.test.js (hunk 4 above)
Comment (salmanm) [reply test/cookie.test.js]: Ok, moved this one out

Comment (jsumners) [reply test/cookie.test.js]: I didn't see an issue. They were subtests covering an overall parent subject.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**